validator/sawtooth_validator/execution/executor.py

```diff
@@ -230,14 +232,19 @@ def __init__(self, service, context_manager, config_view_factory):
         self._executing_threadpool = ThreadPoolExecutor(max_workers=5)
         self._alive_threads = []
         self._lock = threading.Lock()
+        self._scheduler_type = scheduler_type
 
     def create_scheduler(self,
                          squash_handler,
                          first_state_root,
                          always_persist=False):
-        return SerialScheduler(squash_handler=squash_handler,
-                               first_state_hash=first_state_root,
-                               always_persist=always_persist)
+        if self._scheduler_type == "serial":
+            return SerialScheduler(squash_handler=squash_handler,
+                                   first_state_hash=first_state_root,
+                                   always_persist=always_persist)
+        elif self._scheduler_type == "parallel":
+            return ParallelScheduler(squash_handler=squash_handler,
+                                     first_state_hash=first_state_root)
```

Comment:
We should raise an error here in a final else clause.  Probably an `AssertionError`, as it would be programmer error.